appinventor/appengine/src/com/google/appinventor/server/tokenauth/tr.proto

```diff
@@ -0,0 +1,44 @@
+// NOTE WELL!!! This file must define the same structures as the version
+// in the components hierarchy and the version used by the proxy server
+
+// package util;
+
+option java_package = "com.google.appinventor.server.tokenauth";
+option java_outer_classname = "TranslatorToken";
+
+message unsigned {
+    optional string huuid = 1;  // obfuscated userId of requestor
+    optional uint64 version = 2 [default = 0]; // incremented in case of previous revoked
+    optional uint64 generation = 3 [default =0];
+}
+
+// This is the actual message token. The "unsigned" field
+// contains the serialized version of the "unsigned" message
+// above. The "signature" fields contains the raw bytes of
+// the output of HMAC-SHA1 using the key identified by
+// "keyid"
+
+// When stored in a project, this token is base58 encoded
+
+message token {
+    optional uint64 version = 1 [default = 1];
+    optional uint64 keyid = 2 [default = 1];
```

Comment:
I don't see anything wrong here, but I don't understand the reasons for the values of version and generation (or what these really are).  Might be good to add a comment.